### ISSUE TITLE
Semantic Versioning & Changelog

### ISSUE THREAD

**Issue by glumb** [surveyjs/survey-creator#655]
We are currently evaluating the use of SurveyJs/Creator in our product (and  ofc buy a license then).

While managing dozens of dependencies, it is crucial for us to rely on **semantic versioning** of each dependency and a release **changelog**.

For surveyJs/creator its quite hard to figure out what changed for each release without actually digging trough all the code changes. Thus updating the library version is a quite time consuming and labor intensive process.

Because we want to deeply integrate surveyjs in our app many customizations have been made. Updating survey-creator regularly breaks the app and we manually have to figure out what changed and how to fix it.

Don't get me wrong - continuous feature updates and fixes are great, but there must be a way for the integrator to see what changed and if the change affects the implementation.

https://github.com/buefy/buefy/releases has a great way of listing features, fixes and breaking changes

**Comment by andrewtelnov** [comment]
@glumb We will definately introduce something similar. I do not think that we will use GitHub releases, since we have libraries in four repositories: SurveyJS Library, SurveyJS Creator, SurveyJS Pdf Export and Analytics Pack. Some people use them all and it would be inconvinience for them to go to every repo. We will likely do it on our surveyjs.io site. We have to release a new version of this site on next week and we will think about adding a versioning section into it.

Thank you,
Andrew

**Comment by glumb** [comment]
@andrewtelnov that sounds great. Keep it up! 😀

**Comment by andrewtelnov** [comment]
@glumb [Here](https://surveyjs.io/whatsnew) is what we have so far.

Thank you,
Andrew

**Comment by glumb** [comment]
I just upgraded from 1.5.10 to 1.5.18. 
We made some pretty extensive changes to the creator css to match our CI. 
Despite only seeing one breaking change (title rendering), unfortunately, the complete creator looks broken now. Why do you keep changing CSS classes without documenting it?
This constantly breaks all apps of users doing any css customization. Now we can't update and take advantage of bug fixed.
Changing classes of a library providing customizable ui is a breaking change and must be avoided whenever possible.

**Comment by tsv2013** [comment]
@glumb It's hard to discuss a matter without the subject. Can you share a plunker (or other live sample) here?

**Comment by andrewtelnov** [comment]
@glumb I understand your frusturation. Nobody likes to do an extra job that really can be avoided.
In fact we did not inroduce many changes into latest several weeks. However, you are right the markup could be change in several areas. We were working on accessibility issues. 

We have a large enterprise from the states that put it as number one requrement and plus for the last two months we got many new clients related with health care and medical care. Accessibility is a  top requirement for them.

We did a good job regarding accessibility with SurveyJS Library over a year ago, but our SurveyJS Creator had many issues in this area. To be honest, we believed it is not so important and we understand that we had to care about it more. We changed our markup to fix these issues.

We are really trying to be more transporant and we have introduced [what's new section](https://surveyjs.io/WhatsNew) on our website. We will try to make sure that all our changes are included into the list.

Thank you,
Andrew
